plugins/check_curl.c

```diff
@@ -453,7 +463,7 @@ check_http (void)
   int i;
   char *force_host_header = NULL;
   struct curl_slist *host = NULL;
-  char addrstr[100];
+  char addrstr[DEFAULT_BUFFER_SIZE/2];
```

Comment:
I would actually introduce a HOSTNAME_SIZE or so.. The host name should be 64 or so for a machine name and maybe 255 for a FQDN? I'm lacking actual data here..

Reply:
Ah. This is the whole list of hostname.. ok then..

Reply:
Yeah, this is a bit tricky. We need enough space to store all the IPs in `addrstr`, but also enough extra space in the final buffer passed to curl to store the FQDN + port + all IPs. Halving the default buffer size seemed like a quick way to have enough space for both, without hard-coding numbers. 1024 bytes should be enough for up to 25 IPv6 addresses, which i think is enough? In any case it'll just drop any additional IPs that don't fit without failing.

But, feel free to modify the buffer sizes as you see fit if there are more logical values.